ansible/roles/docker/handlers/main.yml

```diff
@@ -1,3 +1,50 @@
 ---
 - name: restart docker
-  service: name=docker state=restarted
+  command: /bin/true
+  notify:
+    - check docker existence
+    - stop docker
+    - delete docker0
+    - start docker
+
+- name: start docker
+  command: /bin/true
+  notify:
+    - check docker existence
+    - docker graph driver
+    - docker fix start on btrfs
+    - docker start service
+
+- name: check docker existence
+  command: "docker version"
+  # if rc!=2 (No such file or directory), assume that docker has installed
+  register: has_docker
+  changed_when: false
+  failed_when: false
+
+- name: stop docker
+  service: name=docker state=stopped
+  when: '"docker0" in ansible_interfaces'
+
+- name: delete docker0
+  command: ip link delete docker0
+  when: '"docker0" in ansible_interfaces'
+
+  # TODO: work only on default installation...
+- name: docker graph driver 
+  command: findmnt -no FSTYPE -T /var/lib/docker
```

Comment:
And obviously docker may fail by any other reason, not only because btrfs. (KO)
We touch part of config here only in case btrfs.
Also, I think we need act in same way in `Turn down docker logging` task (change only `--log-level` or add it if missing).